### DISCUSSION TITLE
Why the `can()` function lowercase the permission parameter ?

### DISCUSSION THREAD

**Discussion by YusufDurmaz** [codeigniter4/shield#1213] [Q&A] [answered]
The `can()` function converts permission to lowercase letters. Why is there such a thing?
The function is looking for featureType.manage permission as featuretype.manage. This took me an hour to unnecessarily review my code.If it is not neceserry which software developer did this and why?

**Comment by datamweb** [comment]
**Improved Performance**: Comparisons become faster and more efficient.
**Uniform Comparisons**: It removes case sensitivity, ensuring all permissions are correctly recognized.
**Better Readability**: It simplifies and standardizes the format, making permissions easier to work with.
**Reduced User Errors**: Mistakes from inconsistent use of uppercase and lowercase letters are eliminated.
**Compatibility with Standards**: Using lowercase aligns with other systems and APIs.
**Simplified Group Management**: Access control works uniformly for all users and groups.

**Reply by YusufDurmaz** [reply]
I don't think there is any advantage in forcibly converting featureType to featuretype. If I wanted to write featuretype anyway, I would. Shield shouldn't force me to do this.

**Reply by kenjis** [reply]
> Shield shouldn't force me to do this.

Shield does. If we remove it, it would be a breaking change. Existing apps may break, so we can't change.

I recommend you to use all lowercase names.
See https://shield.codeigniter.com/references/authorization/#defining-available-permissions

**Reply by YusufDurmaz** [reply]
ok but at least admit that it's not true.

**Reply by lonnieezell** [reply]
It was originally done as a way to help ensure that a typo on the developer's part didn't open up access to an area that someone shouldn't have access to.

**Reply by YusufDurmaz** [reply]
Yes, this makes more sense. However, mistyping can result not only from uppercase or lowercase errors but also from accidentally typing incorrect letters etc. Shield didn't need to do anything for a situation like this.There are already many dangers that can arise from carelessness.

**Reply by datamweb** [reply, marked as answer]
The topic of lowercase conversion in the system is subjective; some developers may see it as a bug while others view it as an advantage. It has been consistently applied in areas like user email storage for consistency. The documentation does not include the example mentioned(featureType.manage), and making changes at this stage (version 1.1.0) could disrupt many users.

@YusufDurmaz Adding a note for users could improve clarity and transparency regarding this decision. If desired, a pull request can be submitted to update the documentation for better accuracy.

**Reply by YusufDurmaz** [reply]
@datamweb Yes, you may be right. It would be good to specify this in the documentation. But i cant create a pull request as i understood

**Reply by YusufDurmaz** [reply]
@datamweb  So will you create a pull request about it ?

**Reply by datamweb** [reply]
@YusufDurmaz I would like you to proceed with this task. Do you have any issues with submitting a pull request?

**Reply by YusufDurmaz** [reply]
@datamweb ok i find out how to do it
